analyzer/src/main/kotlin/managers/Conda.kt

```diff
@@ -0,0 +1,162 @@
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        val workingDir = definitionFile.parentFile
+        val envDir = setupEnv(definitionFile)
+
+        val support = object : PythonSupport(managerName, analysisRoot, command(workingDir), envDir) {
+            override fun runInEnv(workingDir: File, commandName: String, vararg commandArgs: String): ProcessCapture =
+                this@Conda.runInEnv(envDir, commandName, *commandArgs)
+        }
+
+        val result = support.resolveDependencies(definitionFile)
+
+        // Remove the env by simply deleting the directory.
+        envDir.safeDeleteRecursively()
+
+        return result
+    }
+
+    private fun setupEnv(definitionFile: File): File {
+        // Create an out-of-tree environment.
+        log.info { "Creating a Conda env for $definitionFile..." }
+        val envDir = createEnv(definitionFile)
+        val install = installDependencies(definitionFile, envDir)
+
+        if (install.isError) {
+            log.debug {
+                // Pip writes the real error message to stdout instead of stderr.
+                "First try to install dependencies using Conda failed with:\n${install.stdout}"
+            }
+
+            if (install.isError) {
+                // pip writes the real error message to stdout instead of stderr.
+                throw IOException("setupEnv (in $envDir) error: ${install.stdout}")
+            }
+        }
+
+        log.info {
+            "Successfully installed dependencies for project '$definitionFile' using Conda."
+        }
+
+        return envDir
+    }
+
+    /**
+     * Create a Conda environment given the environment.yml or requirements.txt.
+     */
+    fun createEnv(envDefinition: File): File {
+        if (envDefinition.name.endsWith(".py")) {
+            // Python install script is not an environment definition.
+            ProcessCapture(
+                "conda", "create",
+                "-y",
+                "--name", envName,
+                "python"
+            ).requireSuccess()
+        } else {
+            ProcessCapture(
+                "conda", "env", "create",
+                "--force",
+                "--name", envName,
+                "--file", envDefinition.absolutePath
+            ).requireSuccess()
+        }
+
+        // Load the list of Conda environments and find the one we created.
+        val jsonString: String = ProcessCapture("conda", "env", "list", "--json").requireSuccess().stdout
+        val rootNode = jsonMapper.readTree(jsonString)
+        val envsNodes = rootNode.path("envs").elements().asSequence().toList()
+        return File(envsNodes.single { File(it.textValue()).name == envName }.textValue())
+    }
+
+    private fun installDependencies(definitionFile: File, envDir: File): ProcessCapture {
+        // Install pipdeptree inside the Conda environment as that is the only way to make it report only the project's
```

Comment:
The `comment` and code for installing PIPDEPTREE is now duplicated. Would it be reasonable to deduplicate that ?